app/assets/javascripts/discourse/app/controllers/user-private-messages-user.js

```diff
@@ -0,0 +1,52 @@
+import I18n from "I18n";
+import Controller, { inject as controller } from "@ember/controller";
+import { readOnly } from "@ember/object/computed";
+import discourseComputed from "discourse-common/utils/decorators";
+import { inject as service } from "@ember/service";
+
+export default Controller.extend({
+  user: controller(),
+  viewingSelf: readOnly("user.viewingSelf"),
+  router: service(),
+
+  @discourseComputed(
+    "viewingSelf",
+    "router.currentRoute.name",
+    "currentUser.admin"
+  )
+  showWarningsWarning(viewingSelf, currentRouteName, isAdmin) {
+    return (
+      currentRouteName === "userPrivateMessages.user.warnings" &&
+      !viewingSelf &&
+      !isAdmin
+    );
+  },
+
+  @discourseComputed(
+    "pmTopicTrackingState.newIncoming.[]",
+    "pmTopicTrackingState.statesModificationCounter"
+  )
+  newLinkText() {
+    return this._linkText("new");
+  },
+
+  @discourseComputed(
+    "pmTopicTrackingState.newIncoming.[]",
+    "pmTopicTrackingState.statesModificationCounter"
+  )
+  unreadLinkText() {
+    return this._linkText("unread");
+  },
```

Comment:
what would you say about defining some interface user-messages controller which would define functions like:
```
newLinkText() {
    return this._linkText("new");
  },
unreadLinkText() {
    return this._linkText("unread");
  },```

Maybe even `_linkText(type) {` as those functions are quire similar.

In that case, we would have generic controller and individuals would be more like configurations

Reply:
Actually, there are not so similar so probably a wrong idea